app/src/org/commcare/views/widgets/Prototype1.java

```diff
@@ -0,0 +1,148 @@
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
+import org.commcare.dalvik.R;
+import org.javarosa.core.model.data.DateData;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.form.api.FormEntryPrompt;
+
+import java.util.Calendar;
+import java.util.Date;
+
+
+/**
+ * Created by Saumya on 6/1/2016.
+ */
+public class Prototype1 extends QuestionWidget {
+
+    private CalendarWidget myCal;
+    private GregorianDateWidget myGreg;
+    private ImageButton openCalButton;
+
+    public Prototype1(Context con, FormEntryPrompt prompt){
+        super(con, prompt);
+        myGreg = new GregorianDateWidget(con, prompt);
+        myCal = new CalendarWidget(con, prompt);
+
+        myCal.setVisibility(GONE);
+
+        myGreg.removeQuestionText();
+        myCal.removeQuestionText();
+
+        addView(myCal);
+        addView(myGreg);
+
+        initButtons();
+    }
+
+    private void initButtons() {
+        openCalButton = new ImageButton(getContext());
+
+        openCalButton.setImageResource(R.drawable.avatar_vellum_date);
+
+        RelativeLayout layout = (RelativeLayout) findViewById(R.id.widgetinfo);
+        layout.addView(openCalButton);
+        RelativeLayout.LayoutParams params = (RelativeLayout.LayoutParams) openCalButton.getLayoutParams();
+        params.addRule(RelativeLayout.RIGHT_OF, R.id.gregdayofweek);
+        params.width = 60;
+        params.height = 60;
+
+        openCalButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                openCalendar();
+            }
+        });
+
+        ImageButton calendarCloser = new ImageButton(getContext());
+        calendarCloser.setImageResource(R.drawable.close_cross_icon);
+
+        layout = (RelativeLayout) findViewById(R.id.calendarinfo);
+        layout.addView(calendarCloser);
+        params = (RelativeLayout.LayoutParams) calendarCloser.getLayoutParams();
+        params.addRule(RelativeLayout.RIGHT_OF, R.id.calendarweekday);
+        params.width = 90;
+        params.height = 90;
+
+        calendarCloser.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                closeCalendar();
+            }
+        });
+
+        //TODO: Add green "Submit" button. Not done so far because it's unclear what its point is. Navigation to next question is handled already.
+    }
+
+    public ImageButton getCalendarButton(){
+        return openCalButton;
+    }
+
+    private void openCalendar() {
+
+        if(myGreg.getAnswer() != null){
+            myCal.setDate((DateData) myGreg.getAnswer());
+        }else{
+            myCal.setDate(new DateData(new Date()));
+        }
+
+        myGreg.setFocus(getContext());
+        myGreg.setVisibility(GONE);
+        myCal.setVisibility(VISIBLE);
+    }
+
+    private void closeCalendar(){
+        myGreg.setDate((DateData) myCal.getAnswer());

```

Comment:
Interesting - so the gregorian calendar is the final source of truth here? Feels slightly awkward that the CalendarWidget and GregorianWidget both have their own date values that we track and then coalesce. (Bigger comment at PR level)